### ISSUE TITLE
CommandLineToArgvW using old parsing rules

### ISSUE THREAD

**Issue by nu8** [microsoft/Windows-driver-samples#509]
I have seen several projects using the function `CommandLineToArgvW`:

<https://docs.microsoft.com/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw>

another function is available for this purpose, `__p___argv`, which populates
`argv` in C++ and C programs. However it is an internal function:

<https://docs.microsoft.com/cpp/c-runtime-library/internal-crt-globals-and-functions>

which means it is not available to other languages via Foreign function
interface (FFI). Normally this would not be a problem, but since 2008
`__p___argv` and `CommandLineToArgvW` dont follow the same parsing rules:

<http://daviddeley.com/autohotkey/parameters/parameters.htm#WINCRULES>

With the release of the Microsoft C Runtime verion 9, programs that link to
`msvcr90.dll` or higher can utilize the new parsing. To demonstrate the
difference, take this Python program:

~~~py
import sys
s = sys.argv[1]
print(s)
~~~

Example with new parsing:

~~~
C:\> python-2.6.2.amd64\python.exe a.py "a ""b"" c"
a "b" c
~~~

So everything can just be quoted, with an extra Quote to escape quote literals.
Compare with the old style:

~~~
C:\> python-2.5.4.amd64\python.exe a.py "a "\^"b\^"" c"
a "b" c
~~~

Quotes cannot be quoted, and must be double escaped. This has causes issues
with major projects. If the project is not using C++ or C, then they would
basically have to write their own parser to utilize the new rules, as no public
function is currently available that offers it.
